### ISSUE TITLE
Age of frozen beans is misleading

### ISSUE THREAD

**Issue by reox** [graphefruit/Beanconqueror#863]
Frozen beans keep their age at time of freezing. While I can understand the intention, I find it a bit misleading:

* Frozen beans also stale, just much much slower. As this depends on the type of beans, there is possibly also no "freezing-factor" that could be used (I read that freezing is about 10× slower staling on average)
* I maybe want to use the oldest beans first, but without the age information, I have no idea which one that is. Also the freezing date does not help, because I could have bought older beans later.
* The info text says "roasted" - which is then wrong, as it is clearly not anymore the days after roasting...

I would not mind to get back the old behavior, i.e., roasted age really shows the days since roasting, independent of frozen or thawed beans. Another option would be to let the user decide and have an option to toggle or to be able to show both times as a customization setting.

**Comment by graphefruit** [comment]
Hey @reox,
thanks for the ticket.


> * Frozen beans also stale, just much much slower. As this depends on the type of beans, there is possibly also no "freezing-factor" that could be used (I read that freezing is about 10× slower staling on average)

The age is already rightly calculcated, but you'd need to wait for 90 days, that the aging-counter will "rise".
If you want to use the oldest beans, why not sort by roast-date? It would have the same aspect?

> * The info text says "roasted" - which is then wrong, as it is clearly not anymore the days after roasting...

The idea behind this was to not see like "852 days old bean" rather like: You've finished the bean within 30days off roast - if you got a better title, I'm more then happy to tackle this in the next verison.

> Another option would be to let the user decide and have an option to toggle or to be able to show both times as a customization setting.

Maybe two have two fields to display would be the easiest, but still there would be a good naming for it

Have a great cup of coffee and a good slide into 2025 :)
Best
Lars

**Comment by reox** [comment]
> The age is already rightly calculcated, but you'd need to wait for 90 days, that the aging-counter will "rise".

That means right now, the "stale-factor" is 90?

>  If you want to use the oldest beans, why not sort by roast-date? It would have the same aspect?

That is true!

> The idea behind this was to not see like "852 days old bean" rather like: You've finished the bean within 30days off roast - if you got a better title, I'm more then happy to tackle this in the next verison.

Well yeah, I can understand that. I also do not have a good name... I would assume "days since roast" and "days since frozen" would be good - but I don't know about other users' use-cases.

>Have a great cup of coffee and a good slide into 2025 :)

Thanks, you too! And keep up the great work! ☺️

**Comment by graphefruit** [comment]
> That means right now, the "stale-factor" is 90?

Correct. Referencing to the Manchester Coffee Archive: https://manchestercoffeearchive.com/freezing-coffee/


> Well yeah, I can understand that. I also do not have a good name... I would assume "days since roast" and "days since frozen" would be good - but I don't know about other users' use-cases.

Days since frozen would be a special setting just for "frozen" beans, I'll think a bit in it in the future.

**Comment by reox** [comment]
> Correct. Referencing to the Manchester Coffee Archive: https://manchestercoffeearchive.com/freezing-coffee/

interesting! Well, for decaf coffee that is certainly not the case :D Maybe a quick fix would be to have this factor as a setting? In that case I could just set it to 1 to get back the old behavior :D
